internal/db/schema/migrations/oss/postgres/48/03_create_kms_key_destructions.up.sql

```diff
@@ -0,0 +1,40 @@
+begin;
+
+  create table kms_key_destruction_status_enm (
+    name text primary key
+      constraint only_predefined_values_allowed
+      check(name in ('pending', 'running', 'completed', 'failed'))
+  );
+  comment on table kms_key_destruction_status_enm is
+    'Table holding valid statuses a key destruction can take';
+
+  insert into kms_key_destruction_status_enm (name)
+  values
+    ('pending'),
+    ('running'),
+    ('completed'),
+    ('failed');
+
+  create trigger 
+    immutable_columns
+  before
+  update on kms_key_destruction_status_enm
+    for each row execute procedure immutable_columns('name');
+
+  create table kms_key_destruction (
+    private_id wt_private_id primary key,
+    key_id wt_private_id not null, -- Note: not a foreign key, outlives original reference
+    activation_time wt_timestamp not null,
+    deactivation_time wt_timestamp not null,
```

Comment:
What does `deactivation_time` mean? How does it relate to `activation_time`? Can `activation_time` be greater than `deactivation_time`? How does it relate to the `_time` columns? Does it have any relationship with the `state` column?

Reply:
`deactivation_time` is the time the key was deactivated, i.e. when it was rotated and no longer used to encrypt new data, only decrypt historical data. It will always be greater than `activation_time`, as keys can only progress from `deactivated` after being `active`. It does not have any relationship to the `state` column.